### PR TITLE
Content reporting on record fields

### PR DESCRIPTION
**Pull request by dholms**
This extends the work from https://github.com/bluesky-social/atproto/pull/1336 and applies it to record fields.

Namely:
- profile displayName
- list name
- feed generator displayName
- any rkey

Closes https://github.com/bluesky-social/atproto/issues/1328

### CODE REVIEW

File: packages/pds/src/index.ts [hunk 1]
```diff
@@ -175,6 +177,23 @@ export class PDS {
 
     const labelCache = new LabelCache(db)
 
+    let contentReporter: ContentReporter | undefined = undefined
+    if (config.unacceptableHandleWordsB64) {
+      contentReporter = new ContentReporter({
+        backgroundQueue,
+        moderationService: new ModerationService(
```
Comment (dholms) [inline packages/pds/src/index.ts:184]: I don't love how this ended up (instantiating a service here) - but I was having difficulty finding the correct ordering of these 🤔 

I guess one option would be to move the content reporting functionality to the moderation service...

File: packages/pds/src/content-reporter/explicit-slurs.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  /\b[tŤťṪṫŢţṬṭȚțṰṱṮṯŦŧȾⱦƬƭƮʈT̈ẗᵵƫȶ][rŔŕŘřṘṙŖŗȐȑȒȓṚṛṜṝṞṟR̃r̃ɌɍꞦꞧⱤɽᵲᶉꭉ][aÁáÀàĂăẮắẰằẴẵẲẳÂâẤấẦầẪẫẨẩǍǎÅåǺǻÄäǞǟÃãȦȧǠǡĄąĄ́ą́Ą̃ą̃ĀāĀ̀ā̀ẢảȀȁA̋a̋ȂȃẠạẶặẬậḀḁȺⱥꞺꞻᶏẚＡａ4]+[nŃńǸǹŇňÑñṄṅŅņṆṇṊṋṈṉN̈n̈ƝɲŊŋꞐꞑꞤꞥᵰᶇɳȵꬻꬼИиПпＮｎ]{1,2}([iÍíi̇́Ììi̇̀ĬĭÎîǏǐÏïḮḯĨĩi̇̃ĮįĮ́į̇́Į̃į̇̃ĪīĪ̀ī̀ỈỉȈȉI̋i̋ȊȋỊịꞼꞽḬḭƗɨᶖİiIıＩｉ1lĺľļḷḹl̃ḽḻłŀƚꝉⱡɫɬꞎꬷꬸꬹᶅɭȴＬｌ][e3ЄєЕеÉéÈèĔĕÊêẾếỀềỄễỂểÊ̄ê̄Ê̌ê̌ĚěËëẼẽĖėĖ́ė́Ė̃ė̃ȨȩḜḝĘęĘ́ę́Ę̃ę̃ĒēḖḗḔḕẺẻȄȅE̋e̋ȆȇẸẹỆệḘḙḚḛɆɇE̩e̩È̩è̩É̩é̩ᶒⱸꬴꬳＥｅ]|[yÝýỲỳŶŷY̊ẙŸÿỸỹẎẏȲȳỶỷỴỵɎɏƳƴỾỿ]|[e3ЄєЕеÉéÈèĔĕÊêẾếỀềỄễỂểÊ̄ê̄Ê̌ê̌ĚěËëẼẽĖėĖ́ė́Ė̃ė̃ȨȩḜḝĘęĘ́ę́Ę̃ę̃ĒēḖḗḔḕẺẻȄȅE̋e̋ȆȇẸẹỆệḘḙḚḛɆɇE̩e̩È̩è̩É̩é̩ᶒⱸꬴꬳＥｅ][rŔŕŘřṘṙŖŗȐȑȒȓṚṛṜṝṞṟR̃r̃ɌɍꞦꞧⱤɽᵲᶉꭉ])[sŚśṤṥŜŝŠšṦṧṠṡŞşṢṣṨṩȘșS̩s̩ꞨꞩⱾȿꟅʂᶊᵴ]?\b/,
+]
+
+export const hasExplicitSlur = (handle: string): boolean => {
+  return explicitSlurRegexes.some(
+    (reg) => reg.test(handle) || reg.test(handle.replaceAll('.', '')),
```
Comment (devinivy) [inline packages/pds/src/content-reporter/explicit-slurs.ts:15]: For rkey checks in particular, removing `-` may serve a similar purpose to removing `.` from handles.

File: packages/pds/src/repo/prepare.ts [hunk 3]
```diff
@@ -256,3 +259,17 @@ async function cidForSafeRecord(record: RepoRecord) {
     throw badRecordErr
   }
 }
+
+async function assertNoExplicitSlurs(rkey: string, record: RepoRecord) {
```
Comment (devinivy) [inline packages/pds/src/repo/prepare.ts:263]:
```suggestion
function assertNoExplicitSlurs(rkey: string, record: RepoRecord) {
```

File: packages/pds/src/index.ts (hunk 1 above)
Comment (devinivy) [reply packages/pds/src/index.ts:184]: That would def work!  This isn't so bad either—personally not too worried about it either way!

File: packages/pds/src/content-reporter/explicit-slurs.ts (hunk 2 above)
Comment (dholms) [reply packages/pds/src/content-reporter/explicit-slurs.ts:15]: yup good call 👍

**Review by devinivy — APPROVED** [review]

**Pull request merged**